server/handlers/product.js

```diff
@@ -1,80 +1,262 @@
-const Product = require("../models/product");
+const db = require("../models");
+const { acquireCartLock, releaseCartLock } = require("./cartLock");
 
 // Get all products
 const getAllProducts = async (req, res, next) => {
     try {
-        const products = await Product.find();
-        res.status(200).json(products);
+        const products = await db.Product.find();
+        return res.status(200).json(products);
     } catch (err) {
-        res.status(500).json({
-            error: err.message,
+        return next({
+            status: 500,
+            message: err.message,
         });
     }
 };
 
 // Get an product by id
 const getOneProduct = async (req, res, next) => {
     try {
-        const product = await Product.findById(req.params?.id);
-        res.status(200).json(product);
+        const product = await db.Product.findById(req.params?.productId);
+        return res.status(200).json(product);
     } catch (err) {
-        res.status(500).json({
-            error: err.message,
+        return next({
+            status: 500,
+            message: err.message,
         });
     }
 };
 
-// Create an new product
+// Create an new product - only vendor
 const createProduct = async (req, res, next) => {
     try {
-        const product = await Product.create(req.body);
-        await product.save();
-        res.status(201).json({ message: "product created" });
+        const product = await db.Product.create({
+            ...req.body,
+            createdBy: req.params.id,
+        });
+        // Find the vendor and save product into vendor's stock
+        const vendor = await db.User.findById(req.params.id);
+        if (!vendor) {
+            return res.status(401).json({ error: "Vendor do not exist" });
+        }
+        vendor.stock.push(product.id);
```

Comment:
where is the product.id defined?

Reply:
`product.id` will be automatically created when you execute `await db.Product.create()` to create an `Product `instance.

Reply:
> where is the product.id defined?

`product.id` will be automatically created when you execute `await db.Product.create()` to create an `Product` instance.